services/elections.request.service.ts

```diff
@@ -0,0 +1,340 @@
+import { Injectable } from '@angular/core';
+import { HttpClient } from '@angular/common/http';
+import { RequestService } from './request.service';
+import { Observable } from 'rxjs/internal/Observable';
+import { map } from 'rxjs/internal/operators/map';
+import { Election, Position, Candidate, Vote, Ballot, BallotPOST } from 'src/shared-ng/interfaces/elections';
+
+@Injectable()
+export class ElectionsRequestService extends RequestService {
+  baseURL = 'elections';
+
+  constructor(http: HttpClient) {
+    super(http);
+  }
+  ////////////////////
+  // Elections
+  ///////////////////
+  /**
+   * Lists elections
+   *
+   * https://docs.aswwu.com/?url=https://raw.githubusercontent.com/ASWWU-Web/python_server/develop/docs/elections.yml#/election/get_election
+   * @param queryParams
+   * @returns an observable of all elections or elections specified by urlParams
+   */
+  listElection(queryParams?: any): Observable<Election[]> {
+    const electionsObservable = super.get(`${this.baseURL}/election`, queryParams).pipe(
+      map((data: {elections: Election[]}) => data.elections)
+    );
+    return electionsObservable;
+  }
+
+  /**
+   * Read election
+   *
+   * https://docs.aswwu.com/?url=https://raw.githubusercontent.com/ASWWU-Web/python_server/develop/docs/elections.yml#/election/get_election__election_id_
+   * @param queryParams
+   * @return a single election JSON object
+   */
+  readElection(queryParams: any): Observable<Election> {
+    const electionObservable = super.get(`${this.baseURL}/election/${queryParams}`);
+    return electionObservable;
+  }
+
+  /**
+   * Read current election
+   *
+   * https://docs.aswwu.com/?url=https://raw.githubusercontent.com/ASWWU-Web/python_server/develop/docs/elections.yml#/election/get_current
+   * @returns a single election JSON object
+   */
+  readElectionCurrent(): Observable<Election> {
+    const electionObservable = super.get(`${this.baseURL}/current`);
+    return electionObservable;
+  }
+
+  /**
+   * Create elections
+   *
+   * https://docs.aswwu.com/?url=https://raw.githubusercontent.com/ASWWU-Web/python_server/develop/docs/elections.yml#/election/post_election
+   * @param data
+   * @returns JSON object containing the election info created
+   */
+  createElection(data: any): Observable<Election> {
```

Comment:
If we have electionPost type we should use that as an annotation on data instead of any, otherwise this is good
```suggestion
  createElection(data: ElectionPost): Observable<Election> {
```

Reply:
looked into this but I don't think we have that as a type